Tribler/Core/Modules/MetadataStore/OrmBindings/collection_node.py

```diff
@@ -0,0 +1,412 @@
+from __future__ import absolute_import, print_function
+
+import os
+
+from ipv8.database import database_blob
+
+from pony import orm
+from pony.orm import db_session, select
+
+from Tribler.Core.Modules.MetadataStore.OrmBindings.channel_metadata import chunks
+from Tribler.Core.Modules.MetadataStore.OrmBindings.channel_node import (
+    COMMITTED,
+    DIRTY_STATUSES,
+    NEW,
+    TODELETE,
+    UPDATED,
+)
+from Tribler.Core.Modules.MetadataStore.OrmBindings.torrent_metadata import tdef_to_metadata_dict
+from Tribler.Core.Modules.MetadataStore.serialization import COLLECTION_NODE, CollectionNodePayload
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.Utilities.random_utils import random_infohash
+from Tribler.Core.Utilities.unicode import ensure_unicode
+from Tribler.Core.exceptions import DuplicateTorrentFileError
+
+
+def define_binding(db):
+    class CollectionNode(db.MetadataNode):
+        _discriminator_ = COLLECTION_NODE
+
+        # FIXME: ACHTUNG! PONY BUG! attributes inherited from multiple inheritance are not cached!
+        # Therefore, we are forced to move the attributes to common ancestor class of CollectionNode and ChannelTorrent,
+        # that is MetadataNode. When Pony fixes it, we must move it here for clarity.
+        # num_entries = orm.Optional(int, size=64, default=0)
+
+        # Special class-level properties
+        _payload_class = CollectionNodePayload
+        payload_arguments = _payload_class.__init__.__code__.co_varnames[
+            : _payload_class.__init__.func_code.co_argcount
+        ][1:]
+        nonpersonal_attributes = db.MetadataNode.nonpersonal_attributes + ('num_entries',)
+
+        @property
+        @db_session
+        def state(self):
+            # TODO: optimize this by stopping doing blob comparisons on each call, and instead remember rowid?
+            if self.is_personal:
+                return "Personal"
+            return "Preview"
+
+        @db_session
+        def to_simple_dict(self):
+            result = super(CollectionNode, self).to_simple_dict()
+            result.update(
+                {
+                    "torrents": self.num_entries,
+                    "state": self.state,
+                    "total": self.contents_len,
+                    "dirty": self.dirty if self.is_personal else False,
+                }
+            )
+            return result
+
+        def make_copy(self, tgt_parent_id, recursion_depth=15, **kwargs):
```

Comment:
No, we don't. We don't even have a GUI facility to display this kind of information. To show it, we need a blocking dialog window with a progress bar *a la* TotalCommander. Estimating the time to copy can take as much time as the process itself. However, a good estimation can be taken from the channel size itself, since it includes a recursive count of contents.